spec-tests/src/test/java/org/candlepin/spec/activationkey/ActivationKeySpecTest.java

```diff
@@ -0,0 +1,651 @@
+        updatedActivationKey = adminClient.activationKeys()
+            .getActivationKey(activationKey.getId());
+        assertEquals(0, updatedActivationKey.getProducts().size());
+    }
+
+    @Test
+    public void shouldAllowAutoAttachFlagToBeSetOnActivationKeys() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        ActivationKeyDTO activationKey = createActivationKey(adminClient, owner);
+
+        ActivationKeyDTO updatedActivationKey =
+            updateActivationKey(adminClient, activationKey.autoAttach(true));
+        assertThat(updatedActivationKey)
+            .hasFieldOrPropertyWithValue("autoAttach", true);
+        updatedActivationKey = updateActivationKey(adminClient, activationKey.autoAttach(false));
+        assertThat(updatedActivationKey)
+            .hasFieldOrPropertyWithValue("autoAttach", false);
+    }
+
+    @Test
+    public void shouldAllowOverridesToBeAddedToKeys() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        ActivationKeyDTO activationKey = createActivationKey(adminClient, owner);
+        ContentOverrideDTO contentOverride = ContentOverrides.random();
+
+        adminClient.activationKeys().addActivationKeyContentOverrides(
+            activationKey.getId(), List.of(contentOverride));
+        List<ContentOverrideDTO> listOfContentOverride = adminClient.activationKeys()
+            .listActivationKeyContentOverrides(activationKey.getId());
+        assertThat(listOfContentOverride)
+            .hasSize(1)
+            .first()
+            .usingRecursiveComparison()
+            .ignoringFields("created")
+            .ignoringFields("updated")
+            .isEqualTo(contentOverride);
+    }
+
+    @Test
+    public void shouldAllowOverridesToBeRemovedFromKeys() {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        ActivationKeyDTO activationKey = createActivationKey(adminClient, owner);
+        ContentOverrideDTO contentOverride = ContentOverrides.random();
+
+        adminClient.activationKeys().addActivationKeyContentOverrides(
+            activationKey.getId(), List.of(contentOverride));
+        List<ContentOverrideDTO> listOfContentOverride = adminClient.activationKeys()
+            .listActivationKeyContentOverrides(activationKey.getId());
+        assertThat(listOfContentOverride)
+            .hasSize(1);
+        adminClient.activationKeys().deleteActivationKeyContentOverrides(
+            activationKey.getId(), List.of(contentOverride));
+        listOfContentOverride = adminClient.activationKeys()
+            .listActivationKeyContentOverrides(activationKey.getId());
+        assertThat(listOfContentOverride)
+            .hasSize(0);
+    }
+
+    @Test
+    public void shouldAllowNilContentOverrides() {
```

Comment:
We can updated the name from `shouldAllowNilContentOverrides` to `shouldAllowNullContentOverrides` since the test is in java now.

Reply:
done